eclipse/src/saros/ui/widgets/ConnectionStateComposite.java

```diff
@@ -122,7 +100,8 @@ public void widgetDisposed(DisposeEvent e) {
    * @param state the current connection state or <code>null</code>
    * @param error additional error information or <code>null</code>
    */
-  private void updateLabel(ConnectionState state, Exception error) {
+  private void updateLabel(ConnectionState state, ErrorType error) {
+    if (ConnectionStateComposite.this.isDisposed()) return;
```

Comment:
just isDisposed()

It is also questionable if we should do such checks in code that is normally executed when the widget is not disposed.

See ChatRoomsComposite where you did not moved the isDisposed check.

Reply:
Changed it.

Looks a bit superfluous but in some edge cases...

Yes, but didn't wanted to change the other listeners as well.